Build.ps1

```diff
@@ -1,38 +1,79 @@
-$suffix = @{ $true = ""; $false = "$($branch.Substring(0, [math]::Min(10,$branch.Length)))-$revision"}[$branch -eq "main" -and $revision -ne "local"]
-$commitHash = $(git rev-parse --short HEAD)
-$buildSuffix = @{ $true = "$($suffix)-$($commitHash)"; $false = "$($branch)-$($commitHash)" }[$suffix -ne ""]
+    $dbp = [Xml] (Get-Content .\Directory.Version.props)
+    $versionPrefix = $dbp.Project.PropertyGroup.VersionPrefix
 
-Write-Output "build: Package version suffix is $suffix"
-Write-Output "build: Build version suffix is $buildSuffix"
+    Write-Output "build: Package version prefix is $versionPrefix"
 
-& dotnet build --configuration Release --version-suffix=$buildSuffix /p:ContinuousIntegrationBuild=true
+    $branch = @{ $true = $env:CI_TARGET_BRANCH; $false = $(git symbolic-ref --short -q HEAD) }[$NULL -ne $env:CI_TARGET_BRANCH];
+    $revision = @{ $true = "{0:00000}" -f [convert]::ToInt32("0" + $env:CI_BUILD_NUMBER, 10); $false = "local" }[$NULL -ne $env:CI_BUILD_NUMBER];
+    $suffix = @{ $true = ""; $false = "$($branch.Substring(0, [math]::Min(10,$branch.Length)) -replace '([^a-zA-Z0-9\-]*)', '')-$revision"}[$branch -eq "main" -and $revision -ne "local"]
+    $commitHash = $(git rev-parse --short HEAD)
+    $buildSuffix = @{ $true = "$($suffix)-$($commitHash)"; $false = "$($branch)-$($commitHash)" }[$suffix -ne ""]
 
-if($LASTEXITCODE -ne 0) { throw 'build failed' }
+    Write-Output "build: Package version suffix is $suffix"
+    Write-Output "build: Build version suffix is $buildSuffix"
 
-if($suffix) {
-    & dotnet pack src\Serilog.Settings.Configuration --configuration Release --no-build --no-restore -o artifacts --version-suffix=$suffix
-} else {
-    & dotnet pack src\Serilog.Settings.Configuration --configuration Release --no-build --no-restore -o artifacts
-}
+    & dotnet build -c Release --version-suffix=$buildSuffix /p:ContinuousIntegrationBuild=true
+    if($LASTEXITCODE -ne 0) { throw "Build failed" }
+
+    foreach ($src in Get-ChildItem src/*) {
+        Push-Location $src
+
+        Write-Output "build: Packaging project in $src"
+
+        if ($suffix) {
+            & dotnet pack -c Release --no-build --no-restore  -o ../../artifacts --version-suffix=$suffix
+        } else {
+            & dotnet pack -c Release --no-build --no-restore  -o ../../artifacts
+        }
+        if($LASTEXITCODE -ne 0) { throw "Packaging failed" }
 
-if($LASTEXITCODE -ne 0) { throw 'pack failed' }
+        Pop-Location
+    }
 
-Write-Output "build: Testing"
+    foreach ($test in Get-ChildItem test/*.Tests) {
+        Push-Location $test
 
-# Dotnet test doesn't run separate TargetFrameworks in parallel: https://github.com/dotnet/sdk/issues/19147
-# Workaround: use `dotnet test` on dlls directly in order to pass the `--parallel` option to vstest.
-# The _reported_ runtime is wrong but the _actual_ used runtime is correct, see https://github.com/microsoft/vstest/issues/2037#issuecomment-720549173
```

Comment:
.NET 9 changes the default behavior here https://github.com/dotnet/sdk/issues/19147#issuecomment-1935764489